### PR TITLE
feat: Add new pref keys and implement includeProtected notes display criteria 

### PR DESCRIPTION
**Pull request by amanharwara**
1. Adds `NotesShowDeleted` and `NotesHideProtected` pref keys
2. Implements `includeProtected` notes display criteria to show/hide protected notes.
3. Adds a test for the above.

### CODE REVIEW

File: packages/snjs/lib/models/app/userPrefs.ts [hunk 1]
```diff
@@ -31,7 +33,9 @@ export type PrefValue = {
   [PrefKey.SortNotesBy]: CollectionSort;
   [PrefKey.SortNotesReverse]: boolean;
   [PrefKey.NotesShowArchived]: boolean;
+  [PrefKey.NotesShowDeleted]: boolean;
```
Comment (mobitar) [inline packages/snjs/lib/models/app/userPrefs.ts:36]: Hmm looks like we already have something similar? https://github.com/standardnotes/snjs/pull/457/files#diff-bbe8e7705d98fb9ec1181e74bc0c8bb20e12c3ac1b8596494c6c56a696de40fcR101

Comment (mobitar) [reply packages/snjs/lib/models/app/userPrefs.ts:36]: Line 101
Screen Shot 2021-10-17 at 10 11 53 AM

Comment (mobitar) [reply packages/snjs/lib/models/app/userPrefs.ts:36]: Ok, let's rename this to Show Trashed at the very least, as Deleted is a separate concept that is permanent and irrecoverable.

Comment (amanharwara) [reply packages/snjs/lib/models/app/userPrefs.ts]: The design says "Show deleted notes". I guess I should change that as well?

**Comment by amanharwara** [comment]
Version numbers follow semver right?

**Pull request merged**